### PR TITLE
memory: increase heap buffer size for TGL platform

### PR DESCRIPTION
**Pull request by mrajwa**
This patch increases HEAP_BUFFER_SIZE so more advanced pipelines
can be allocated. Also HEAP_BUFFER_BLOCK_SIZE has been decreased
to allocate buffers with better granuality

Signed-off-by: Marcin Rajwa <marcin.rajwa@linux.intel.com>

### CODE REVIEW

File: src/platform/tigerlake/include/platform/lib/memory.h [hunk 1]
```diff
@@ -263,8 +263,8 @@
 	HEAP_RT_COUNT1024 * 1024 + HEAP_RT_COUNT2048 * 2048 + \
 	HEAP_RT_COUNT4096 * 4096)
 
-#define HEAP_BUFFER_SIZE	0xF000
-#define HEAP_BUFFER_BLOCK_SIZE		0x180
+#define HEAP_BUFFER_SIZE	0x50000
+#define HEAP_BUFFER_BLOCK_SIZE		0x100
```
Comment (lgirdwood) [inline src/platform/tigerlake/include/platform/lib/memory.h]: We should make the block size consistent across all platforms

Comment (paulstelian97) [reply src/platform/tigerlake/include/platform/lib/memory.h]: Also is there enough memory for that significant increase (from 60k to 320k)?

Comment (lbetlej) [reply src/platform/tigerlake/include/platform/lib/memory.h]: @paulstelian97 AFAIK there is still memory left as TGL has ~3MB local SRAM. @mrajwa Do you know what is approx. safety margin you have with this HEAP_BUFFER_SIZE for requested typologies ?

Comment (mrajwa) [reply src/platform/tigerlake/include/platform/lib/memory.h]: @lbetlej, yes this PR changes the HEAP_BUFFER_SIZE to reasonable size which is about 10% of whole memory and allows us to load multi pipeline topology including WoV. We still have plenty of room available so there is no reason to worry about boundaries at this point of time as we are not even close to them.

Comment (paulstelian97) [reply src/platform/tigerlake/include/platform/lib/memory.h]: That's what I wanted to hear, that there is a 3MB total.

On i.MX8* I have an 8MB range for the heaps. It's got a few blocks (I'll add a few 4kB blocks soon on one of the platforms) and everything else in that 8MB range will be buffer heaps. Essentially the memory map already has reserved a large enough chunk that there isn't need for more in the foreseeable future. When I need something in a heap other than the buffer heap, I just move blocks around.

**Comment by lgirdwood** [comment]
Jenkins known issue.

**Comment by lyakh** [comment]
@lgirdwood @mrajwa I think there's a problem to this PR. It looks like 0x180 wasn't a random number, it was selected because "typical" (I guess you can derive this from our "typical" sampling rate, format, number of channels and ALSA behaviour) audio allocation sizes are 0x180 and 0x300. After this PR has changed that to 0x100 now to allocate 0x180 you need two 0x100-byte buffers, thus wasting 128 bytes for each such allocation. Maybe buffers of 0x80 bytes would actually be the best choice.

**Comment by lgirdwood** [comment]
@lyakh can you check the impact here of aligning the buffer block granularity to 0x80 - it may grow the heap data.

**Comment by lyakh** [comment]
> @lyakh can you check the impact here of aligning the buffer block granularity to 0x80 - it may grow the heap data.

@lgirdwood let me push such a draft patch for BYT, let's see how it passes tests

**Pull request merged**